interview-questions.md

```diff
@@ -8,50 +8,92 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+Researched answer: In my original answer, I described iteration as being used in for loops. I have since learned that iteration and looping are similar, related, yet distinct concepts, although they are often used interchangeably. Both looping and iteration will perform a certain code repeatedly, however iteration refers most commonly to when that process is done just once for each item in an an array or string, until it reaches the end, and looping refers to doing that process until a certain condition is met. In a JavaScript for loop, this is defined within the second statement of the for loop. i.e., "continue looping until this condition is met." Iteration typically has a default ending point which does not need to be defined in the code (i.e. the end of the array). Running through the code once could be described as one execution of a loop, or one iteration, so it's easy to see how these terms are close to synonymous.
 
-Researched answer:
 
-5. What is the difference between a class and an object?
 
-Your answer:
+5. What is the difference between a class and an object?
 
-Researched answer:
+Your answer: Objects are collections of key:value pairs which are stored inside curly braces. Class is a function which is used to create a template, or blueprint, for making objects. Classes can also includes methods, which are functions contained within and belonging to that class. For example, if I needed to create many objects such as products in an online shop, I could first create a class called Product which would use the method constructor to take parameters of the defining properties all of the products would have, such as brand, type, price, category, etc. Then, to create the objects, I could use the keyword class to quickly and easily set up many different objects.
+
+class Product = {
+    constructor(brand, type, price, category){
+        this.brand = brand
+        this.type = type
+        this.price = price
+        this.category = category
+    }
+    describe() = {return `In the ${this.category} section, we have a ${this.brand} {this.type} for the low price of $${this.price}!`
+}
+let keurig = new Product("Keurig", "coffee maker", "65", "Applicances")
+let hottub = new Product("Whirlpool", "jacuzzi", "1500", "Outdoor")
+
+Researched answer: When learning more about classes, I learned about the idea of hoisting. This refers to what order JS reads the code and performs it. While some functions can be invoked before they are actually defined, classes do not fall into this category. First, a class must be declared, and then, below, they can be used to construct objects.
+I also learned about class expressions which are a way of defining classes. For example, if there are going to be several similar but distinct classes, it may be helpful to group them within a class expression. The syntax would be:
```

Comment:
There are other ways to connect similar classes that we will expand on when we get into Ruby called class inheritance.  Javascript does have this and is a really great concept to dig into if you are curious about connecting similar classes and passing attributes from a parent class to a child class (similar to what we did with props in React).  [Class Inheritance](https://javascript.info/class-inheritance)